python/pyspark/sql/tests.py

```diff
@@ -894,6 +894,13 @@ def test_capture_analysis_exception(self):
         # RuntimeException should not be captured
         self.assertRaises(py4j.protocol.Py4JJavaError, lambda: self.sqlCtx.sql("abc"))
 
+    def test_capture_illegalargument_exception(self):
+        self.assertRaises(IllegalArgumentException,
+                          lambda: self.sqlCtx.sql("SET mapred.reduce.tasks=-1"))
```

Comment:
can the test case make sure the error message is still present? It would be bad if we only show IllegalArgumentException, but no error message anymore.